src/main/java/seedu/address/model/entry/Name.java

```diff
@@ -5,26 +5,30 @@
 import seedu.address.commons.exceptions.IllegalValueException;
 
 /**
- * Represents an Entry's name in the entry book.
- * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
+ * Represents an Entry's name in the entry book. Guarantees: immutable; is valid
+ * as declared in {@link #isValidName(String)}
  */
 public class Name {
 
-    public static final String MESSAGE_NAME_CONSTRAINTS =
-            "Entry names should only contain alphanumeric characters and spaces, and it should not be blank";
+    public static final String MESSAGE_NAME_CONSTRAINTS = "Entry names should only start "
+            + "with alphanumeric characters and it should not be blank, "
+            + "subsequent spaces and non-alphanumeric characters are allowed.";
 
     /*
-     * The first character of the entry must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
+     * The first character of the entry must not be a whitespace, otherwise " "
+     * (a blank string) becomes a valid input.
      */
-    public static final String NAME_VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    // public static final String NAME_VALIDATION_REGEX =
+    // "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String NAME_VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum}\\p{S}\\p{P} ]*";
 
     public final String fullName;
 
     /**
      * Validates given name.
      *
-     * @throws IllegalValueException if given name string is invalid.
+     * @throws IllegalValueException
+     *             if given name string is invalid.
```

Comment:
might want to look into why this keeps happening, could be the eclipse autoformatter